### ISSUE TITLE
Install nested packages with relative dependencies

### ISSUE THREAD

**Issue by ierezell** [python-poetry/poetry#4403]
- [X] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Linux 5.13.10-arch1-1  Archlinux

- **Poetry version**: 1.1.7

## Issue
Hello, and thanks in advance for any help. 

I have a structure like this : 

Each of my project have a venv folder in .venv at root and `in-project=false`
```
Root/ 
| .venv/ 
|        --- VenvProj1
|        --- VenvProj2
|        --- VenvProj3
|----- Libs 
|-----Projs
|          ----- Proj1
|                   ​ ----- pyproject.toml
|          ----- Proj2
|                     ----- pyproject.toml
|           ----- Proj3
|                     ----- pyproject.toml
```

However I wanted  all my commands from all projects in a single  place  (plus the config/tests files etc...)
So I added a `pyproject.toml` at root with : `proj1 = { path = "./projs/Proj1/", develop=true}` (etc.. for all others)

Proj2 and proj3 depends on Proj1 and libs with `proj1 = { path = "../Proj1/", develop=true}` (in `proj23/poetry.toml`)

I can do a `poetry install/build` inside each project and publish them (perfect). 
But when I do an install at root, the first one works (with `develop on` and I got all my commands in a poetry shell) by when running the install/update of root a second time it fails as in the lock file the dependencies kept the relative path. 

In the root lock file: proj2 depends on "../proj1" (relative to proj2 with works perfectly inside proj2) but when updating or doing operations from root, then I got a "../proj1" doesn't exist (as it's relative to proj2 and not root).

I also tried with `develop=false` and got the same error directly (the first root install even didn't work). 

Is there a way to convert paths to absolute or relative to root when installing instead of keeping the same?
Else could we install all the subprojects inside the same virtualenv to access all commands when developing ?

A hack would be to build and install proj1 with pip to have access system-wide but is not editable.  

I hope my problem was well stated and not too obscure, ask any questions if not. 
Have a great day

**Comment by finswimmer** [comment]
Closing as possible duplicate of #2270
